diff --git a/rustls/src/client/handy.rs b/rustls/src/client/handy.rs
--- a/rustls/src/client/handy.rs
+++ b/rustls/src/client/handy.rs
@@ -214,9 +214,7 @@ mod preshared_keys {
         ///
         /// Replaces any existing keys for that server name.
         pub fn update(&mut self, server_name: ServerName<'static>, keys: Arc<[PresharedKey]>) {
-            self.keys
-                .entry(server_name)
-                .and_modify(|value| *value = keys);
+            self.keys.insert(server_name, keys);
         }
     }
 
